Shared/Samples/Snap geometry edits/SnapGeometryEditsView.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+/// A view that shows how to interact with the geometry editor with snapping.
+struct SnapGeometryEditsView: View {
+    /// The map to display in the view.
+    @State private var map: Map = {
+        let map = Map(
+            item: PortalItem(
+                portal: .arcGISOnline(connection: .anonymous),
+                // A stripped down Naperville water distribution network webmap.
+                id: PortalItem.ID("b95fe18073bc4f7788f0375af2bb445e")!
+            )
+        )
+        // Snapping is used to maintain data integrity between different sources
+        // of data when editing, so full resolution is needed for valid snapping.
+        map.loadSettings.featureTilingMode = .enabledWithFullResolutionWhenSupported
+        return map
+    }()
+    
+    /// The model that is required by the geometry editor menu.
+    @StateObject private var model = GeometryEditorModel(
+        geometryEditor: GeometryEditor(),
+        graphicsOverlay: GraphicsOverlay(renderingMode: .dynamic)
+    )
```

Comment:
```suggestion
    @StateObject private var model = GeometryEditorModel()
```

Is there a reasons to create these objects here? Seems like it would be simpler to create them in the initalizer.

Reply:
Copy pasted that from the Create and edit geometry sample. I tried to modify as few things as possible to avoid any confusion when we need to refactor both samples. But this looks like an innocuous change so I'll make it.

<sub>some 👿 soul yelling in the distance</sub> - <sup>it's not [dry](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) ;-)</sup>